legend-pure-core/legend-pure-m3-core/src/main/java/org/finos/legend/pure/m3/compiler/postprocessing/processor/valuespecification/FunctionExpressionProcessor.java

```diff
@@ -639,6 +643,26 @@ private static boolean processLambda(FunctionExpression functionExpression, Proc
         return lambdaParametersInferenceSuccess;
     }
 
+    private static void handleTypeArgumentTypeInference(GenericType templateReturnType, GenericType concreteGenericType, TypeInferenceObserver observer, ProcessorState state)
```

Comment:
You should also consider the case when templateTypeArgument is concrete but its raw type is a FunctionType or RelationType.  There can be type parameters nested within a FunctionType or RelationType.

Reply:
We will add these as we see its need.  Currently the lack of the support for these is not blocking usage.